### PR TITLE
search: avoid computing large value of ElapsedMilliseconds for search alert result

### LINKED ISSUE #10701: Large value of ElapsedMilliseconds when empty search returns instantly
Some searches (ones that typically have no results and return instsantly) have a very large `ElapsedMilliseconds` value. This can be reproduced against the sourcegraph.com or local endpoint with a [`src`](https://github.com/sourcegraph/src-cli) query like:

```
SRC_ENDPOINT=https://sourcegraph.com  SRC_ACCESS_TOKEN="<your-token>" src search -json 'repo:*'
```

And you'll see:

```
{
  "SourcegraphEndpoint": "https://sourcegraph.com",
  "Query": "repo:*",
  "Site": {
    "BuildVersion": "63706_2020-05-14_48cb2bc"
  },
  "Results": [],
  "LimitHit": false,
  "Cloning": [],
  "Missing": [],
  "Timedout": [],
  "ResultCount": 0,
  "ElapsedMilliseconds": 2077252342
}
```

This is not an issue in `src`, it is reproducible in the api console, so the problem lies in the backend code:

Screen Shot 2020-05-14 at 5 20 37 PM

I think I've seen this recurrently in the last couple of months, so I don't think it's a new issue. This is likely what affects some metric logging #10058. Something is going awry, and it is likely an integer conversion problem, so the code and values that contribute to [calculating ElapsedMilliseconds](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@48cb2bc0136f768ff18577ab4c6a6e11b431cdb8/-/blob/cmd/frontend/graphqlbackend/search_results.go#L195) needs investigating.

### PR DESCRIPTION
**Pull request by fedealconada**
This PR  fixes the large value of ElapsedMilliseconds when empty search returns instantly

Fixes #10701

### CODE REVIEW

**Comment by fedealconada** [comment]
@rvantonder this would be option 1) solution.

**Comment by rvantonder** [comment]
This seems fine as a workaround. Could we write a test that just covers this change for search alert result?

**Comment by fedealconada** [comment]
@rvantonder do you think a test like the one I have just pushed would work?

**Comment by rvantonder** [comment]
If using `return &SearchResultsResolver{alert: alert}` instead of the change we're introducing here causes the test to fail, then yes, it is strong enough.

File: cmd/frontend/graphqlbackend/search_alert.go [hunk 1]
```diff
@@ -598,7 +598,7 @@ func (a searchAlert) Results(context.Context) (*SearchResultsResolver, error) {
 		description:     a.description,
 		proposedQueries: a.proposedQueries,
 	}
-	return &SearchResultsResolver{alert: alert}, nil
+	return &SearchResultsResolver{alert: alert, start: time.Now()}, nil
```
Comment (rvantonder) [inline cmd/frontend/graphqlbackend/search_alert.go:601]: Could you add a sentence or two commenting why this is necessary (i.e., what the workaround does).

Comment (fedealconada) [reply cmd/frontend/graphqlbackend/search_alert.go:602]: what about now?

File: cmd/frontend/graphqlbackend/search_alert.go [hunk 2]
```diff
@@ -598,7 +598,8 @@ func (a searchAlert) Results(context.Context) (*SearchResultsResolver, error) {
 		description:     a.description,
 		proposedQueries: a.proposedQueries,
 	}
-	return &SearchResultsResolver{alert: alert}, nil
+	// Passing time.now() is a workaround that prevents (sr *SearchResultsResolver) ElapsedMilliseconds() from returning a wrong calculation
```
Comment (rvantonder) [inline cmd/frontend/graphqlbackend/search_alert.go:601]:
```suggestion
	// ElapsedMilliseconds() will calculate a very large value for duration if start takes on the nil-value of year 1. As a workaround, instantiate start with time.now(). TODO(rvantonder): #10801.
```

File: cmd/frontend/graphqlbackend/search_alert.go (hunk 1 above)
Comment (rvantonder) [reply cmd/frontend/graphqlbackend/search_alert.go:602]: all good

File: cmd/frontend/graphqlbackend/search_alert.go (hunk 2 above)
Comment (fedealconada) [reply cmd/frontend/graphqlbackend/search_alert.go]: Much better, thanks!

**Pull request merged**